### PR TITLE
Ruleset save button & other

### PR DESCRIPTION
**Pull request by Scott-Code-19**
## Describe Your Changes
Multiple other Stories were Implemented into this branch 
#80 #81 #82 #83 

- Updated README
- Implemented 90% of the UI
- Added Base Angular Routing
- Implemented UI functionality
  - Save Button
  - Cancel Button #82 
  - Move up & Move Down 
- Implemented Backend functionality
  - GetByID Endpoint #80 
  - DeleteByID Endpoint #81 
  - Put Endpoint #83

## Type of Change
- [X] New feature (change that adds functionality)

## Checklist (DOD)
- [X] The code is written and passes all Acceptance criteria's
- [X] Any Zombie code is removed
- [X] Code has proper comment documentation
- [X] The code is unit tested with happy case and passes
- [X] Code is refactored for optimization
- [x] The code has been peer-reviewed by one or more team members who did not work on it.
- [x] The code has no merge conflicts

### CODE REVIEW

**Comment by Scott-Code-19** [comment]
Fixing more things right now

File: front-end/src/app/cancel-button/cancel-button.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser'
+import { CancelButtonComponent } from './cancel-button.component';
+import * as rxjs from 'rxjs';
```
Comment (GivenTeacher5) [inline front-end/src/app/cancel-button/cancel-button.component.spec.ts:4]: unused import
remove if not needed
clarify if needed

**Review by sophialhales — CHANGES_REQUESTED** [review]
Looks good, requested some changes

File: front-end/src/app/app.component.css [hunk 2]
```diff
@@ -2,7 +2,9 @@ main {
     height:100vh;
 }
 
-app-sidebar {
+.sidebar {
+    position: relative;
```
[the remark this answers is not in the document]
Comment (Scott-Code-19) [reply front-end/src/app/app.component.css:6]: I'm going to leave all 3 but it seems that only.
position: sticky;
top: 0;
are needed

File: src/main/java/com/accesspoint/rulesengine/entity/Condition.java [hunk 3]
```diff
@@ -31,10 +31,12 @@ public class Condition implements Serializable {
...
 
     @JsonIgnore
-    @ManyToOne
+    @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "rule_id", nullable=false, updatable=false)
     @ColumnTransformer(write = "?::bigint")
     @EqualsAndHashCode.Exclude
     @ToString.Exclude
     private Rule rule;
+
```
[the remark this answers is not in the document]
Comment (GivenTeacher5) [reply src/main/java/com/accesspoint/rulesengine/entity/Condition.java:40]: im not sure what you mean by this. looks fine to me

File: front-end/src/app/cancel-button/cancel-button.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Scott-Code-19) [reply front-end/src/app/cancel-button/cancel-button.component.spec.ts:4]: removed with Zach's comment

File: src/main/java/com/accesspoint/rulesengine/controller/RulesetController.java [hunk 4]
```diff
@@ -20,12 +20,23 @@ public List<RulesetModel> all() {
...
     }
 
+    @GetMapping("/ruleset/{id}")
+    public ResponseEntity<Ruleset> one(@PathVariable Long id) {
+        return rulesetService.getById(id);
+    }
+
     @PostMapping("/ruleset")
-    ResponseEntity<Ruleset> newRuleset(@RequestBody CreateRuleSetRequest request) {
+    ResponseEntity<Ruleset> newRuleset(@RequestBody Ruleset request) {
```
[the remark this answers is not in the document]
Comment (Scott-Code-19) [reply src/main/java/com/accesspoint/rulesengine/controller/RulesetController.java:29]: Done

**Review by sophialhales — APPROVED** [review]
Looks good to me!

**Review by GivenTeacher5 — APPROVED** [review]
mmm good

**Pull request merged**